chore(contribution.md): adds definiton of done (#2422)
swisspost/design-system: CONTRIBUTING.md

| `pnpm header:start`               | starts the stencil compiler for the header                                                   |
| `pnpm icons:start`                | starts the http server for debugging downloaded icons                                        |

> #### Definition of Done for a Story

Reviewer: ```suggestion
### Definition of Done for a Story
```

> - Component docs

Reviewer: Could you convert all the top-level items (Component docs, Controls, Testing, etc.) to a h4 (e.g. #### Component docs)?

  - Short introduction
  - Canvas for showcasing
  - Controls for the showcase
  - Variants if applicable (e.g. group of component or small example in context)
  - Instructions for installation
- Controls

>   - Possibility to re-create all variants (good for debugging because they can be set via query params and a reproduction can be linked)

Reviewer 2: ```suggestion
  - Possibility to showcase and test all variants of the component. This enables re-usage of the docs programmatically for visual regression testing.
```

  - Accessibility hints for relevant controls like disabled state or hidden labels
- Testing

> 

Reviewer: ```suggestion
```

>   - Cross browser testing of the docs

Reviewer 2: ```suggestion
  - Cross browser testing of the docs (official Post supported browsers -> https://github.com/swisspost/design-system/blob/main/packages/styles/.browserslistrc)
```

  - Testing in High Contrast mode

>     - Try emulating the high contrast mode in chrome by yourself:

Reviewer 2: ```suggestion
    - Emulating the high contrast mode in chrome:
```

      - Open the Command Menu: `ctrl` + `shift` + `P` (or `cmd` + `shift` + `P` on mac).
      - Type `rendering` and press `Enter` .
      - In the Rendering panel, scroll down to the Emulate CSS media feature forced-colors and activate it from the drop-down.
      - You can switch also prefers-color-scheme to test light/dark mode.
  - Approval by UX for the story

>   - For more complex widgets: screen reader testing (at least nvda)

Reviewer 2: ```suggestion
  - Screen reader testing (at least nvda)
```

- Accessibility
  - `aria-hidden="true"` or span.visually-hidden labels for all icons
- Docs

>   - No `react related attributes in code snippets (e.g. className, htmlFor, key, etc.)

Reviewer 2: ```suggestion
  - No react related attributes in code snippets (e.g. className, htmlFor, key, etc.)
```

>   - No nested p's (beware of .lead and .alert)

Reviewer 2: ```suggestion
  - No nested `<p>` tags (beware of .lead and .alert)
```

>   - When using LinkTo use both kind and story attributes

Reviewer 2: ```suggestion
  - When using LinkTo use both the `kind` and `story` attributes
```

>   - `div.hide-col-default` wrapper for Controls component on CSS only docs

Reviewer 2: ```suggestion
  - `div.hide-col-default` wrapper for controls on CSS only component docs
```

>   - Add the `sourceState="shown"` attribute to the first canvas if the code is not too long.

Reviewer 2: ```suggestion
  - Add the `sourceState="shown"` attribute to the first canvas if the code is not too long (less than 8 lines)
```

- Naming convention
  - Docs file: `[component].docs.mdx`
  - Story file: `[component].stories.ts`
  - Demo story file: `[component].demo.stories.tsx`
  - Snapshot story file: `[component].snapshot.stories.ts`

>   - style file `[component].styles.scss`

Reviewer 2: ```suggestion
  - Style file `[component].styles.scss`
```

  - SCSS export file `[component].module.scss`
  - Custom mdx helper `blocks [component].blocks.tsx`
- General Coding Rules
  - Use already existing utilities instead of custom solutions.

>     For example use mapClasses utility function instead of `[*].filter(c => c && c !== 'null').join(' ')`

Reviewer 2: ```suggestion
    For example use the `mapClasses` utility function instead of `[*].filter(c => c && c !== 'null').join(' ')`.
```

| `pnpm header:start`               | starts the stencil compiler for the header                                                   |
| `pnpm icons:start`                | starts the http server for debugging downloaded icons                                        |

### Definition of Done for a Story

#### Component docs

- Short introduction
- Canvas for showcasing
- Controls for the showcase
- Variants if applicable (e.g. group of component or small example in context)
- Instructions for installation

#### Controls

- Possibility to showcase and test all variants of the component. This enables re-usage of the docs programmatically for visual regression testing.
- Accessibility hints for relevant controls like disabled state or hidden labels

#### Testing

> - Cross browser testing of the docs (official Post supported browsers -> https://github.com/swisspost/design-system/blob/main/packages/styles/.browserslistrc)

Reviewer 3: Maybe this is something to be reviewed since: https://github.com/swisspost/design-system/issues/2425

> - Testing in High Contrast mode

Reviewer 3: ```suggestion
- Testing the High Contrast mode
```

  - Emulating the high contrast mode in chrome:
    - Open the Command Menu: `ctrl` + `shift` + `P` (or `cmd` + `shift` + `P` on mac).
    - Type `rendering` and press `Enter` .
    - In the Rendering panel, scroll down to the Emulate CSS media feature forced-colors and activate it from the drop-down.

>     - You can switch also prefers-color-scheme to test light/dark mode.

Reviewer 3: This is not something that I'm expecting to be written here as I'm not a chrome user and you don't want to list all possibilities.

- Approval by UX for the story

> - Screen reader testing (at least nvda)

Reviewer 3: ```suggestion
- Screen reader testing (at least nvda+win)
```
In the screen reader context, the combination of OS + screen reader matters.

#### Accessibility

> - `aria-hidden="true"` or span.visually-hidden labels for all icons

Reviewer 3: ```suggestion
- `aria-hidden="true"` or `span.visually-hidden` labels for all icons
```
